core/blockchain.go

```diff
@@ -1414,9 +1459,27 @@ func (bc *BlockChain) reorgNeeded(localBlock *types.Block, localTd *big.Int, ext
 	return reorg
 }
 
+// pruneBlockSidecars prunes the sidecars of blocks that are older than the keep period
+func (bc *BlockChain) pruneBlockSidecars(db ethdb.KeyValueWriter, curBlock *types.Block) {
```

Comment:
I think we should think about this carefully. It might happen the case when a node sync data from other node, what if other node is malicious, they send ancient block together with its sidecars (self-saved), and broadcast to us, we do save these ancient sidecars, and actually persist here permanently here.